python_ta/reporters/html_reporter.py

```diff
@@ -84,21 +86,51 @@ def output_blob(self):
         # Date/time (24 hour time) format:
         # Generated: ShortDay. ShortMonth. PaddedDay LongYear, Hour:Min:Sec
         dt = str(datetime.now().strftime('%a. %b. %d %Y, %I:%M:%S %p'))
-        output_path = os.path.join(os.getcwd(), self.linter.config.pyta_output_file)
+
+        # Render the jinja template
+        rendered_template = template.render(date_time=dt, reporter=self)
+
+        # If a filepath was specified, write to the file
+        if self._output_filepath:
+            self._write_html_to_file(rendered_template)
+        else:
+            self._open_html_in_browser(rendered_template.encode('utf8'))
```

Comment:
I'm thinking it might be best to do the `.encode('utf8')` call for both kinds of outputs, which means you could add it to Line 91 above.

You can also test this yourself by running PythonTA on a file where there's an error on a line with Unicode characters (you can include such characters in variable names, for example).

Reply:
okay, the only thing I was scared of originally not to pass in a `.encode`d version of `rendered_template` to `self._write_html_to_file` was because the original code to write to the file already had something that handled encoding and I did not want to modify too much of the old functionality and code.

```python
    def _write_html_to_file(self, rendered_template):
        output_path = os.path.join(os.getcwd(), self.linter.config.pyta_output_file)
        with open(output_path, 'w', encoding='utf-8') as f:
            f.write(rendered_template)
```

would I need to change anything to the function if I passed in an `.encode('utf8')`d version?

Reply:
@simonchenwastaken you should test it yourself, in the way that I described.